webviz_subsurface/_providers/ensemble_summary_provider/_provider_impl_arrow_lazy.py

```diff
@@ -123,17 +134,26 @@ class Elapsed:
         timer = PerfTimer()
 
         unique_column_names = set()
-        for table in per_real_tables.values():
+        for real_num, table in per_real_tables.items():
             unique_column_names.update(table.schema.names)
 
             if "REAL" in table.schema.names:
-                raise ValueError("Input tables should not have REAL column")
+                raise ValueError(
+                    f"Input tables should not have REAL column (real={real_num})"
+                )
 
             if table.schema.field("DATE").type != pa.timestamp("ms"):
-                raise ValueError("DATE column must have timestamp[ms] data type")
-
-            if not _is_date_column_sorted(table):
-                raise ValueError("DATE column must be sorted")
+                raise ValueError(
+                    f"DATE column must have timestamp[ms] data type (real={real_num})"
+                )
+
+            if not _is_date_column_monotonically_increasing(table):
+                offending_pair = _find_first_non_increasing_date_pair(table)
+                raise ValueError(
+                    f"DATE column must be monotonically increasing\n"
+                    f"Error detected in realization: {real_num}\n"
+                    f"First offending timestamps: {offending_pair[0], offending_pair[0]}"
```

Comment:
```suggestion
                    f"First offending timestamps: {offending_pair}"
```
Would result in the same message? Saves a few letters of code 😆